src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -345,11 +349,22 @@ private Task ProcessNegotiate(HttpContext context)
             }
         }
 
-        private async Task<bool> EnsureConnectionStateAsync(ConnectionState connectionState, HttpContext context, string transportName)
+        private async Task<bool> EnsureConnectionStateAsync(ConnectionState connectionState, HttpContext context, TransportType transportType, TransportType supportedTransports)
         {
+            var transportName = transportType.ToString().ToLower();
+
+            if ((supportedTransports & transportType) == 0)
+            {
+                context.Response.StatusCode = StatusCodes.Status404NotFound;
+                // REVIEW: Is this information leakage? Should we only show this in "development"?
+                await context.Response.WriteAsync($"{transportName} transport not supported by this end point type");
```

Comment:
When we have negotiate that returns supported transports - would it be different?